tests/CppWinRTTests.cpp

```diff
@@ -26,6 +27,79 @@ static const HRESULT cppwinrt_mapped_hresults[] =
     E_OUTOFMEMORY,
 };
 
+template<typename T> auto copy_thing(T const& src)
+{
+    return std::decay_t<T>(src);
+}
+
+template<typename T, typename K> 
+void CheckMapVector(std::vector<winrt::Windows::Foundation::Collections::IKeyValuePair<T, K>> const& test, std::map<T, K> const& src)
+{
+    REQUIRE(test.size() == src.size());
+    for (auto&& i : test)
+    {
+        REQUIRE(i.Value() == src.at(i.Key()));
+    }
+}
+

```

Comment:
Maybe add cases for things like `DeviceInformationCollection` and `ResourceMapIterator`.

```cpp
struct vectorLike : implements<vectorLike, IVectorView<int>>
{
    auto Size() { return 0; }
    int GetAt(uint32_t) { throw hresult_out_of_bounds() }
    uint32_t GetMany(uint32_t start, array_view<int>) { if (start > 0) throw hresult_out_of_bounds(); return 0; }
    bool IndexOf(int, uint32_t) { throw hresult_not_implemented(); }
};
```